Update guide for v43. (#16832)
ckeditor/ckeditor5: docs/updating/update-to-43.md

## Update to CKEditor&nbsp;5 v43.0.0

_Released on August 7, 2024._

For the entire list of changes introduced in version 43.0.0, see the [release notes for CKEditor&nbsp;5 v43.0.0](https://github.com/ckeditor/ckeditor5/releases/tag/v43.0.0).

Below are the most important changes that require your attention when upgrading to CKEditor&nbsp;5 v43.0.0.

### Export to Word v2 becomes the default

> In version v42.0.0, we introduced new version of Export to Word plugin that utulizes updated v2 of the converter. It was introduced in an opt-in manner. From the v43.0.0, the Export to Word v2 is **the default** configuration. This may require updates to the editor's configuration if you use this plugin.

Reviewer: ```suggestion
In version v42.0.0, we introduced a new version of the Export to Word plugin that utilizes the updated v2 of the converter. It was introduced in an opt-in manner. From v43.0.0, the Export to Word v2 is **the default** configuration. This may require updates to the editor's configuration if you use this plugin.
```

> In CKEditor&nbsp;5 v42.0.0, we introduced a new version of Export to Word plugin that utilizes the updated converter (marked as V2). It was introduced in an opt-in manner. Starting with v43.0.0, the Export to Word V2 is **the default** configuration. This may require updates to the editor's configuration if you use this plugin.

Reviewer 2: We use vXX.x.x in CKE5, but EtW uses V1/V2 (capitalized).

Example v1 configuration:

```js
exportWord: {
            tokenUrl: 'https://example.com/cs-token-endpoint',
            fileName: 'my-file.docx',
            converterOptions: {
                format: 'A4',
                margin_top: '20mm',
                margin_bottom: '20mm',
                margin_right: '12mm',
                margin_left: '12mm'
            }

>         }

Reviewer: Should those snippets be nested?

```suggestion
exportWord: {
    tokenUrl: 'https://example.com/cs-token-endpoint',
    fileName: 'my-file.docx',
    converterOptions: {
        document: {
            size: 'A4',
            margin: {
                top: '20mm',
                bottom: '20mm',
                right: '12mm',
                left: '12mm'
            }
        }
    }
}
```

```

The same migrated to V2:

```js
exportWord: {
			tokenUrl: 'https://example.com/cs-token-endpoint',
			fileName: 'my-file.docx',
			converterOptions: {
				document: {
					size: 'A4',
					margin: {
						top: '20mm',
						bottom: '20mm',
						right: '12mm',
						left: '12mm'
					}
				}
			}
		}
```

You can find the full V2 configuration in the {@link module:export-word/exportword~ExportWordConverterOptionsV2} API documentation.

Alongside with the configuration changes we decided to switch the default `auto_pagination` option to `false`. This option was only used with the pagination feature. When it is enabled the breaks in the pagination force the page breaks in the exported Word document. However, browser engines and Microsoft Word differ significantly. Because of that we decided to turn this option to `false`. If you still want to enforce breaks from pagination, change it in the configuration to `true`.

> ### Typing bug fixes

Reviewer 2: Seldom do we mention such stuff in the migration guide.

Author: We discussed on the daily :blush:

> Alongside with the configuration changes we decided to switch the default `auto_pagination` option to `false`. This option was only used along our `Pagination` feature. When this option is enabled the breaks in the pagination force the breaks in the exported Word document. Browser engines and Microsoft Word differ significantly. Because of that we decided to turn this option to `false` If you still want to enforce breaks from pagination, change it in the configuration to `true`.

Reviewer: ```suggestion
Alongside the configuration changes, we switched the default `auto_pagination` option to `false`. We use the new default along with our `Pagination` feature. When this option is enabled, the breaks in the pagination force the breaks in the exported Word document. Browser engines and Microsoft Word differ significantly. Because of that, we decided to turn this option to `false`. If you still want to enforce breaks from pagination, change it in the configuration to `true`.
```

### React and Vue integrations updates

We have released new major versions of the React and Vue integrations. In both of them, we have migrated to JavaScript modules (ESM) and rewritten large parts of their codebases to support the latest versions of these frameworks and to follow the latest recommendations for writing the components.

We strongly recommend that you read the release highlights to update to the latest versions:

- [Release highlights for React integration](https://github.com/ckeditor/ckeditor5-react/releases/tag/v9.0.0).
- [Release highlights for Vue integration](https://github.com/ckeditor/ckeditor5-vue/releases/tag/v7.0.0).

### Updates to the Package Generator

In the last major release of CKEditor&nbsp;5 we introduced new installation methods. Since then we have received a lot of feedback from you. Thanks to this, we identified a number of minor issues fixed in versions 42.0.1 and 42.0.2. While these fixes did not require any changes in the application, they do require some changes in the custom plugins created with the Package Generator.

> You can skip this section if you have not yet read the [Migrating custom plugins](https://ckeditor.com/docs/ckeditor5/latest/updating/nim-migration/custom-plugins.html) guide. However, if you have, follow the steps below to apply these fixes to your plugin.

Reviewer: ```suggestion
You can skip this section if you have not read the [Migrating custom plugins](https://ckeditor.com/docs/ckeditor5/latest/updating/nim-migration/custom-plugins.html) guide. However, if you have, follow the steps below to apply these fixes to your plugin.
```

> In the last major release of CKEditor we introduced new installation methods, and since then we have received a lot of feedback from you. Thanks to this feedback, we identified a number of minor issues that we fixed in versions 42.0.1 and 42.0.2. While these fixes did not require any changes in the consuming application, they do require some changes in the custom plugins created with the Package Generator.

Reviewer: ```suggestion
In the last major release of CKEditor, we introduced new installation methods, and since then, we have received a lot of feedback from you. Thanks to this feedback, we identified several minor issues we fixed in versions 42.0.1 and 42.0.2. While these fixes did not require any changes in the consuming application, they do require some changes in the custom plugins created with the Package Generator.
```

> 1. Update all packages starting with `@ckeditor/ckeditor5-dev-` to version `^42.0.0`.

Reviewer 2: Is the version right here?

Author: Copied from @filipsobol's notes. @pszczesniak?

Reviewer 3: Yes `42.0.0` is the correct version for `ckeditor5-dev-*` packages.
